.github/workflows/build.yml

```diff
@@ -148,8 +146,41 @@ jobs:
           components: rustfmt
       - run: cargo +nightly fmt --all -- --check
 
-  codespell:
-    name: Codespell
+  check-clippy:
```

Comment:
(Optional) If you are doing this much reorganizing, perhaps you could split the `check-` checks into a separate file, `.github/workflow/check.yml`? This file is already getting to long, and the `build-` prefix before this jobs doesn't help much.

<img width="530" alt="image" src="https://github.com/user-attachments/assets/7253cb95-d02f-4c0e-a629-7fc68b6ca69d" />

Reply:
I think that this goes against what I want, though. I want all the information exactly on one page; often when I'm looking at things like build times or errors from builds I want to flip between different things, or I accidentally click on the wrong thing, and that's because they get put in separate workflows (so GitHub's UX shows them as different "groups".)

These checks haven't changed drastically in quite a while and expand slowly; keeping all the information exactly on one page is what I want here. I think renaming things to `test` or something is a good idea though; I'll try to think of something more informative maybe.

Reply:
I renamed this to `ci` actually, since I think that's much more on-point for what we're actually trying to convey with this workflow (all of these need to pass to continuously integrate changes into trunk.) I think that's actually quite good and it keeps the build matrix entries shorter.

Reply:
ci is definitely better than build, thanks! I'll trust you on not splitting them, my instinct is different, but I haven't thought about it as much.